cluster/swarm/cluster.go

```diff
@@ -255,7 +262,7 @@ func (c *Cluster) listNodes() []*node.Node {
 	c.RLock()
 	defer c.RUnlock()
 
-	out := []*node.Node{}
+	out := make([]*node.Node, 0, len(c.engines))
```

Comment:
What's the point of optimizing this?

If it's for readability concerns, I guess we could either use the original version, or if we use `make` we shouldn't append but rather do something like `out[i] = n`